analytics_data_api/v0/views/__init__.py

```diff
@@ -97,3 +108,119 @@ def finalize_response(self, request, response, *args, **kwargs):
+
+        GET /api/v0/some_endpoint/?ids={id},{id}
+            Returns list of serialized models with IDs that match an ID in the given
+            `ids` query parameter with all default fields.
+
+        GET /api/v0/some_endpoint/?ids={id},{id}&fields={some_field},{some_field}
+            Returns list of serialized models with IDs that match an ID in the given
+            `ids` query parameter with only the fields in the given `fields` query parameter.
+
+        GET /api/v0/some_endpoint/?ids={id},{id}&exclude={some_field},{some_field}
+            Returns list of serialized models with IDs that match an ID in the given
+            `ids` query parameter with all fields except those in the given `exclude` query
+            parameter.
+
+    **Response Values**
+
+        Since this is an abstract class, this view just returns an empty list.
+
+    **Parameters**
+
+        This view supports filtering the results by a given list of IDs. It also supports
+        explicitly specifying the fields to include in each result with `fields` as well of
+        the fields to exclude with `exclude`.
+
+        ids -- The comma-separated list of identifiers for which results are filtered to.
+            For example, 'edX/DemoX/Demo_Course,course-v1:edX+DemoX+Demo_2016'. Default is to
+            return all courses.
+        fields -- The comma-separated fields to return in the response.
+            For example, 'course_id,created'. Default is to return all fields.
+        exclude -- The comma-separated fields to exclude in the response.
+            For example, 'course_id,created'. Default is to not exclude any fields.
+    """
+    ids = None
+    fields = None
+    exclude = None
+    always_exclude = []
+    serializer_class = serializers.ModelSerializer
+    model = models.Model
+    model_id = 'id'
+
+    def get_serializer(self, *args, **kwargs):
+        kwargs.update({
+            'context': self.get_serializer_context(),
+            'fields': self.fields,
+            'exclude': self.exclude
+        })
+        return self.get_serializer_class()(*args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        query_params = self.request.query_params
+        self.fields = split_query_argument(query_params.get('fields'))
+        exclude = split_query_argument(query_params.get('exclude'))
+        self.exclude = self.always_exclude + (exclude if exclude else [])
+        self.ids = split_query_argument(query_params.get('ids'))
+        self.verify_ids()
+
+        return super(APIListView, self).get(request, *args, **kwargs)
+
+    def verify_ids(self):
+        pass  # to be implemented in subclasses
+
+    def default_result(self, item_id):
+        """Default result with fields pre-populated to default values."""
+        result = {
+            self.model_id: item_id,
+        }
+        return result
+
+    def get_result_from_model(self, model, base_result=None, field_list=None):
```

Comment:
I was having trouble understanding this initially and I think it's because it's more than getting the "results," but about updating it.  `update_result_from_model` instead?